lib/SILGen/SILGenType.cpp

```diff
@@ -1080,7 +1080,7 @@ void SILGenModule::emitNonCopyableTypeDeinitTable(NominalTypeDecl *nom) {
   auto serialized = IsSerialized_t::IsNotSerialized;
   bool nomIsPublic = nom->getEffectiveAccess() >= AccessLevel::Public;
   // We only serialize the deinit if the type is public and not resilient.
-  if (nomIsPublic && !nom->isResilient())
+  if (nomIsPublic && !nom->hasPackageAccess() && !nom->isResilient())
```

Comment:
is it possible that `getEffectiveAccess() == Public && nom->hasPackageAccess()`

Reply:
Yes. This PR https://github.com/apple/swift/pull/69709 changed the effective access level of package declarations to `public` since those declarations are visible across module boundaries.

Reply:
Would it be too disruptive to have getEffectiveAccess() return a new enum that does not include 'Package' then? Or maybe move the computation to the SIL linkage level, so that AccessLevel::Package becomes SILLinkage::Public? getEffectiveAccess() is a terrible hack for `-enable-testing` and we should not hang new behavior off of it. There's also a FormalLinkage concept for type metadata in IRGen, IIRC. It's all too convoluted.